core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -142,6 +145,67 @@ public function getFilePath($absolute = false): string
         return $absolute ? $path : Path::makeRelative($path, $this->projectDir);
     }
 
+    /**
+     * Synchronously process images if they are deferred.
+     *
+     * This will make sure that the target files physically exist instead of
+     * being generated by the Contao\CoreBundle\Controller\ImagesController
+     * on first access.
+     */
+    public function createIfDeferred(): void
+    {
+        $picture = $this->getPicture();
+
+        $array_flatten = static function (array $array): array {
+            return array_reduce($array, 'array_merge', []);
+        };
+
+        $findDeferredImages = static function (array $item) use ($array_flatten): array {
+            $candidates = array_merge(
+                [$item['src'] ?? null],
+                $array_flatten($item['srcset'] ?? [])
+            );
+
+            return array_filter(
+                $candidates,
+                static function ($image): bool {
+                    return $image instanceof DeferredImageInterface;
+                }
+            );
+        };
+
+        // Create an array of deferred images by looking through the 'src' and
+        // 'srcset' attributes of both the 'img' and all 'sources' tags.
+        $deferredImages = array_merge(
+            $findDeferredImages($picture->getImg()),
+            $array_flatten(array_map($findDeferredImages, $picture->getSources()))
+        );
```

Comment:
```suggestion
        $deferredImages = [];
        
        foreach ([$picture->getImg(), ...$picture->getSources()] as $source) {
            $deferredImages[] = $source['src'];
            foreach ($source['srcset'] ?? [] as $srcset) {
            	$deferredImages[] = $srcset[0];
            }
        }
```

Would be shorter and more easy to understand?

Reply:
:smile: Guess so. Besides that `...$picture->getSources()` only works in PHP7.4. But that's easily adjusted.

Is it safe to assume that a `src` key exists and it's always the first element in `$srcset` that contains the image?

Reply:
I just saw that I removed the `instanceof DeferredImageInterface` check 🙈
With the instanceof check every incorrect value would be skipped automatically, we could use `$source['src'] ?? null` and `$srcset[0] ?? null` to allow malformed array structures.

Reply:
Ah yes, just read your comment. That's exactly what I have done in 4f30e2f9f8764de8f1ace572695d3c1e8cdb05a5. :smiley: